### PR TITLE
plugins:dac_data_manager: Add support for devices without DDS

### PR DESCRIPTION
**Pull request by rbolboac**
In case the DAC device does not offer DDS support, but it offers buffer capabilities, show DAC buffer settings.

### CODE REVIEW

**Review by dNechita — APPROVED** [review]
Looks good!

File: plugins/dac_data_manager.c [hunk 1]
```diff
@@ -2035,13 +2036,15 @@ struct dac_data_manager *dac_data_manager_new(struct iio_device *dac,
 void dac_data_manager_free(struct dac_data_manager *manager)
 {
 	if (manager) {
+		if (manager->tones_count) {
 		free(manager->dac1.txs);
 		free(manager->dac2.txs);
+		g_slist_free(manager->dds_tones);
```
Comment (nunojsa) [inline plugins/dac_data_manager.c:2042]: the indentation here is wrong...

File: plugins/dac_data_manager.c [hunk 2]
```diff
@@ -1927,7 +1927,7 @@ static int dds_dac_init(struct dac_data_manager *manager,
 	manager->dacs_count++;
 	ddac->index = manager->dacs_count;
 
-	return ret;
+	return 0;
```
Comment (nunojsa) [inline plugins/dac_data_manager.c:1930]: As this api stands now, there's no point in returning anything... I'm not seeing a path where an error is set. Hence, I'm also not seeing the point of the `fprintf()`...

In fact, I'm wondering if we should have a better way of detecting such devices... Like this, we loose the ability to say that something is wrong for devices that are capable of more than 1 tone

**Comment by rbolboac** [comment]
- fixed indentation 
- changed dds_dac_init

File: plugins/dac_data_manager.c (hunk 2 above)
Comment (dNechita) [reply plugins/dac_data_manager.c:1934]: yes, you are right that we need a better way of detecting devices that do not have DDS.
One criteria would be that the number of tones is equal to 0.

Comment (rbolboac) [reply plugins/dac_data_manager.c:1934]: Thank you for pointing this out!
I made some changes. Do you think the new variant is more suitable?

**Review by nunojsa — APPROVED** [review]

**Pull request merged**